modules/javafx.graphics/src/main/java/javafx/css/SimpleSelector.java

```diff
@@ -76,32 +77,82 @@ public String getName() {
     /**
      * Gets an immutable list of style-classes of the {@code Selector}.
      * @return an immutable list of style-classes of the {@code Selector}
+     * @deprecated use {@link #getStyleClassNames()} instead
      */
+    @Deprecated(forRemoval = true)
     public List<String> getStyleClasses() {
+        return List.copyOf(selectorStyleClassNames);
+    }
+
+    /**
+     * Gets the immutable {@code Set} of {@code StyleClass}es of the {@code Selector}.
+     * @return the {@code Set} of {@code StyleClass}es
+     * @deprecated {@code StyleClass} wrapper has become unnecessary, use {@link #getStyleClassNames()} instead
+     */
+    @Deprecated(forRemoval = true)
+    public Set<StyleClass> getStyleClassSet() {
```

Comment:
Given that the class itself is deprecated for removal, marking this method as deprecated is unnecessary.